src/Components/ChatContainer/ChatContainer.js

```diff
@@ -12,34 +12,27 @@ const endPoint = "http://localhost:5000";  
 function ChatContainer({ groupCode, roomCode, name }) {
   const socket = io.connect(`${endPoint}`);
   const [messages, setMessages] = useState([]);
-  const [roomData, setRoomData] = useState('');
   const [error, setError] = useState('');
   const [roomDetails, setRoomDetails] = useState( { user_two: null } );
 
 
 
-  const getMessages = () => { 
-    socket.on("message", msg => { 
-      setMessages([...messages, JSON.parse(msg)]); 
-    }); 
-  };
-
   useEffect(() => {
     socket.on("message",function(msg) {  
-      getMessages(msg);
+      setMessages([...messages, msg]); 
       socket.emit('recived', true)
     });
   },[messages.length])
 
   useEffect(() => {
     socket.on("join_room",function(data) {  
-      setRoomData(data)
+      setRoomDetails(data)
       socket.emit('recived', true)
     });
-  },[roomData])
+  },[roomDetails])
 
   const leaveChat = () => {
-    socket.emit('leave', roomData)
+    socket.emit('leave', roomDetails)
```

Comment:
necessary?

Reply:
Both of us on separate branches declared a roomDetails/roomData state so I just got rid of one?